src/lib/components/NavMenu.svelte

```diff
@@ -15,17 +15,49 @@
 		clickLogout: void;
 	}>();
 
-	export let conversations: Array<{
+	interface Conv {
 		id: string;
 		title: string;
-	}> = [];
+		updatedAt: Date;
+	}
+	export let conversations: Array<Conv> = [];
 
 	export let canLogin: boolean;
 	export let user: LayoutData["user"];
 
 	function handleNewChatClick() {
 		isAborted.set(true);
 	}
+
+	// group conversations based on conv.updatedAt property which is a date
+	// it should have the following groups: "Today", "This week", "This month", "Older"
+
+	$: groupedConversations = conversations.reduce(
```

Comment:
Reduce is too complex here imo, I would just use filter, something like `.filter(isToday)` in the view